csm_web/scheduler/management/commands/checkintegrity.py

```diff
@@ -44,124 +26,31 @@ def _err(self, msg):
-                        section, section.current_student_count, section.capacity
-                    )
-                )
-            if not section.default_spacetime:
-                self._err("Section {} lacks spacetime".format(section))
-            st = section.default_spacetime
-            obj = (
-                st.start_time,
-                st.location,
-                st.day_of_week,
-                section.mentor.user.email,
-            )
-            if obj in seen_spacetimes:
-                self._err(
-                    "Duplicate spacetime for section {} by mentor {}".format(
-                        section, section.mentor
-                    )
-                )
-            seen_spacetimes.add(obj)
-            if not section.mentor:
-                self._err("Section {} lacks mentor".format(section))
-            mentor_role = section.mentor.role
-            if section.mentor.role == Profile.STUDENT:
-                self._err(
-                    "Somehow, mentor {} of section {} has role student".format(
-                        section.mentor, section
-                    )
-                )
-
-    def _check_profile_integrities(self):
-        for profile in Profile.objects.all():
-            if not profile.user:
-                self._err("Somehow, profile {} has no user".format(profile))
-            if profile.role not in Profile.ROLE_MAP:
-                self._err("Profile {} has bad role".format(profile))
-            if profile.role == Profile.STUDENT and not profile.section:
-                self._err("Student profile {} has no section".format(profile))
-            if profile.role == Profile.STUDENT and profile.active:
-                if profile.leader != profile.section.leader:
-                    self._err(
-                        "Student profile {} has inconsistent leader".format(profile)
-                    )
-
-    def _check_attendance_integrities(self):
-        qs = Attendance.objects.exclude(section=F("attendee__section"))
-        if qs.count() > 0:
-            self._err(
-                "Found {} attendances where section does not match profile".format(
-                    qs.count()
-                )
+            if section.mentor is None:
+                self._err(f"Section {section} has no mentor")
+
+        mentors = Mentor.objects.select_related("section__spacetime")
+        emails = mentors.values_list("user__email", flat=True).distinct()
+        for email in emails:
+            spacetimes = Spacetime.objects.filter(
+                pk__in=mentors.filter(user__email=email).values_list("section__spacetime", flat=True)
             )
+            # Naive n^2 algorithm since n is almost certainly < 2
+            n = len(spacetimes)
+            for i in range(n):
+                st = spacetimes[i]
+                st_start = dt.datetime.combine(dt.date.today(), st.start_time)
+                st_end = st_start + st.duration
+                for j in range(i + 1, n):
```

Comment:
```suggestion
                for st_2 in spacetimes[i+1:]:
```

If you *really* care about the overhead of creating new lists by slicing, see `itertools.islice`